hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -224,54 +224,49 @@ public void join() throws InterruptedException {
-        );
-      } catch (SCMException scmEx) {
-        LOG.warn("Fail to delete block: {}", keyBlocks.getGroupID(), scmEx);
-        AUDIT.logWriteFailure(
-            buildAuditMessageForFailure(SCMAction.DELETE_KEY_BLOCK, auditMap,
-                scmEx)
-        );
-        switch (scmEx.getResult()) {
-        case SAFE_MODE_EXCEPTION:
-          resultCode = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
-              .Result.safeMode;
-          break;
-        case FAILED_TO_FIND_BLOCK:
-          resultCode = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
-              .Result.errorNotFound;
-          break;
-        default:
-          resultCode = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
-              .Result.unknownFailure;
-        }
-      } catch (IOException ex) {
-        LOG.warn("Fail to delete blocks for object key: {}", keyBlocks
-            .getGroupID(), ex);
-        AUDIT.logWriteFailure(
-            buildAuditMessageForFailure(SCMAction.DELETE_KEY_BLOCK, auditMap,
-                ex)
-        );
-        resultCode = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
-            .Result.unknownFailure;
+    ScmBlockLocationProtocolProtos.DeleteScmBlockResult.Result codeResult;
+    try{
+      scm.getScmBlockManager().deleteBlocks(keyBlocksInfoList);
+      codeResult = ScmBlockLocationProtocolProtos.
+              DeleteScmBlockResult.Result.success;
+      AUDIT.logReadSuccess(buildAuditMessageForSuccess(
+              SCMAction.DELETE_KEY_BLOCK, auditMap));
+    } catch (SCMException scmEx){
+      LOG.warn("Fail to delete {} keys", keyBlocksInfoList.size(), scmEx);
+      AUDIT.logWriteFailure(
+              buildAuditMessageForFailure(SCMAction.DELETE_KEY_BLOCK, auditMap,
+                      scmEx)
+      );
+      switch (scmEx.getResult()) {
+      case SAFE_MODE_EXCEPTION:
+        codeResult = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
+                .Result.safeMode;
+        break;
+      case FAILED_TO_FIND_BLOCK:
+        codeResult = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
+                .Result.errorNotFound;
+        break;
+      default:
+        codeResult = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
+                .Result.unknownFailure;
       }
-      List<DeleteBlockResult> blockResultList = new ArrayList<>();
-      for (BlockID blockKey : keyBlocks.getBlockIDList()) {
-        blockResultList.add(new DeleteBlockResult(blockKey, resultCode));
+    } catch (IOException ex){
+      LOG.warn("Fail to delete {} keys", keyBlocksInfoList.size(), ex);
+      AUDIT.logWriteFailure(
+              buildAuditMessageForFailure(SCMAction.DELETE_KEY_BLOCK, auditMap,
+                      ex)
+      );
+      codeResult = ScmBlockLocationProtocolProtos.DeleteScmBlockResult
+              .Result.unknownFailure;
+    }
+    for(BlockGroup bg : keyBlocksInfoList){
+      auditMap.put("KeyBlockToDelete", bg.toString());
```

Comment:
auditMap is used in the audit log messages. So we will have to populate it before the audit log calls.

Reply:
Please refer to the new changes.